lebedigital/openbis/expstep.py

```diff
@@ -0,0 +1,1464 @@
+        props_list = list(o.get_sample_type(sample_type)
+                          .get_property_assignments()
+                          .df['propertyType'])
+
+        # Build a dataframe with the properties and their descriptions
+        df = pd.DataFrame()
+        for prop in props_list:
+            pt = o.get_property_type(prop)
+            prop_series = pd.Series(pt.attrs.all())
+            prop_df = pd.DataFrame(
+                {prop: prop_series.values}, index=prop_series.index)
+
+            if df.empty:
+                df = prop_df
+            else:
+                df = df.join(prop_df)
+
+        prop_df = df
+
+        meta_df['Label'] = prop_df.loc['label'].values
+        meta_df['Description'] = prop_df.loc['description'].values
+
+        # Add an empty column where you can input the values
+        meta_df = meta_df.assign(Value='')
+
+        # If write was specified then write to file
+        if write:
+            # If sheet exists remove it before writing the template
+            if os.path.exists(path):
+                os.remove(path)
+
+            # Write the sheet
+            with pd.ExcelWriter(path=path, engine='xlsxwriter') as writer:
+                meta_df.to_excel(excel_writer=writer, sheet_name=sheet_name)
+
+        else:
+            return meta_df
+
+    def import_from_template(self, path_to_file: str):
+        """Imports the data from the template into the samples metadata. To used with the generated import template
+
+        Args:
+            path_to_file (str): Path to the filled out import template
+        """
+
+        df = pd.read_excel(path_to_file)
+        df.columns = df.columns.str.lower()
+
+        # if some params were left out then the row gets dropped
+        df = df.dropna()
+
+        # not useful for uploading to the datastore only for the user to know what to input
+        df.pop('label')
+        df.pop('description')
+
+        # turn the df columns into a dict
+        meta = dict(zip([par.lower() for par in df.param], df.value))
+        self.metadata = meta
+
+    # The simple static methods will get integrated into their methods later because they are useless to define over pybis itself, self explainatory fucntions otherwise
+    @staticmethod
+    def get_space_names(o: Openbis):
+        spaces_df = o.get_spaces().df
+        return list(spaces_df['code'].values)
+
+    @staticmethod
+    def get_project_names(o: Openbis, space: str):
+        projects_df = o.get_projects(space=space).df
+        return [name.split('/')[-1] for name in list(projects_df['identifier'].values)]
+
+    @staticmethod
+    def get_experiment_names(o: Openbis, space: str, project: str):
+        experiments_df = o.get_experiments(
+            space=space,
+            project=project,
+        ).df
+        return [name.split('/')[-1] for name in list(experiments_df['identifier'].values)]
+
+    # Alias
+    get_collection_names = get_experiment_names
```

Comment:
why do we need this alias?

Reply:
The alias stems from openBIS itself. Collection = experiment, sample = object. I wanted to include them too but i abandoned the idea later and use collection and sample mostly. Ive deleted the aliases